spec/features/visitors/email_confirmation_spec.rb

```diff
@@ -0,0 +1,132 @@
+require 'rails_helper'
+
+feature 'Email confirmation during sign up' do
+  scenario 'confirms valid email and sets valid password' do
+    email = 'test@example.com'
+    sign_up_with(email)
+    raw_confirmation_token, = Devise.token_generator.generate(User, :confirmation_token)
+    User.find_by(email: email).update(
+      confirmation_token: raw_confirmation_token, confirmation_sent_at: Time.current
+    )
+
+    visit "/users/confirmation?confirmation_token=#{raw_confirmation_token}"
+
```

Comment:
Yes we do have access to the email. While looking for the token would be good, I do like Moncef's idea of actually clicking on the link and the verifying that the user is confirmed afterward. Feels more feature spec-y.